pkg/virtctl/imageupload/imageupload.go

```diff
@@ -926,3 +946,57 @@ func handleEventErrors(client kubecli.KubevirtClient, pvcName, dvName, namespace
 
 	return nil
 }
+
+func createNewDataSource(client kubecli.KubevirtClient, name, namespace string) error {
+	labels := make(map[string]string)
+	setDefaultInstancetypeLabels(labels)
+
+	ds := &v1beta1.DataSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: v1beta1.DataSourceSpec{
+			Source: v1beta1.DataSourceSource{
+				PVC: &v1beta1.DataVolumeSourcePVC{
+					Name:      name,
+					Namespace: namespace,
+				},
+			},
+		},
+	}
+
+	_, err := client.CdiClient().CdiV1beta1().DataSources(namespace).Create(context.Background(), ds, metav1.CreateOptions{})
+
+	fmt.Printf("Created new DataSource %s/%s\n", namespace, name)
+	return err
+}
+
+func updateExistingDataSource(client kubecli.KubevirtClient, pvcName, pvcNamespace string, ds *cdiv1.DataSource) error {
```

Comment:
I see, in that case it suffices to pass only the existing labels to this func.

Reply:
Let's leave it as it is. I think it makes sense to have `ds *cdiv1.DataSource` passed to this function because it already exists.